.github/workflows/rust_tests.yml

```
@@ -0,0 +1,114 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# To learn more about GitHub Actions in Apache Beam check the CI.md
+
+name: Rust Tests
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "10 2 * * *"
+  push:
+    branches: ["master", "release-*", "rust_sdk"]
+    tags: ["v*"]
+  pull_request:
+    branches: ["master", "release-*", "rust_sdk"]
+    tags: ["v*"]
+    paths: ["sdks/rust/**"]
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: "${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}"
+  cancel-in-progress: true
+
+jobs:
+  rust_unit_tests:
+    name: "Rust Unit Tests"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        # TODO: the test cannot be compiled under Windows platform, add window-latest and fix the build
```

[nits] The execution of the test fails (not the compilation). I'll commit the small suggestion.

```suggestion
        # TODO: the test fails under Windows platform (for directory separator in `--skip target/debug`. add window-latest and fix the build
```